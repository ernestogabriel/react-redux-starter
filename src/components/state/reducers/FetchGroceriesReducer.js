import {
    FETCH_GROCERIES_PENDING,
    FETCH_GROCERIES_FULFILLED,
    FETCH_GROCERIES_REJECTED,
    FETCH_GROCERY_PENDING,
    FETCH_GROCERY_FULFILLED,
    FETCH_GROCERY_REJECTED
} from '../actions/GroceryActions';

// INITIALIZE STATE

export const initialGroceryCollectionState = {
    groceries: [],
    fetching: false,
    fetched: false,
    failed: false
};

export const initialGroceryState = {
    grocery: null,
    fetching: false,
    fetched: false,
    failed: false
};


export const FetchGroceryReducer = (state = initialGroceryState, action) => {
    switch(action.type){
        case FETCH_GROCERY_PENDING:
            return {
                ...state,
                grocery: null,
                fetching: true,
                fetched: false,
                failed: false
            };
        case FETCH_GROCERY_FULFILLED:
            return {
                ...state,
                grocery: action.payload,
                fetching: false,
                fetched: true,
                failed: false
            };
        case FETCH_GROCERY_REJECTED:
            return {
                ...state,
                groceries: null,
                fetching: false,
                fetched: false,
                failed: true
            };
        default:
            return state;
    }
};
export const FetchGroceriesReducer = (state = initialGroceryCollectionState, action) => {
    switch(action.type) {
        case FETCH_GROCERIES_PENDING:
            return {
                ...state,
                groceries: [],
                fetching: true,
                fetched: false,
                failed: false
            };
        case FETCH_GROCERIES_FULFILLED:
            return {
                ...state,
                groceries: action.payload,
                fetching: false,
                fetched: true,
                failed: false
            };
        case FETCH_GROCERIES_REJECTED:
            return {
                ...state,
                groceries: [],
                fetching: false,
                fetched: false,
                failed: true
            };
        default:
            return state;
    }
};