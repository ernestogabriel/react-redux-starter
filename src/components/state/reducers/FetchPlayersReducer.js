import {
    FETCH_PLAYERS_PENDING,
    FETCH_PLAYERS_FULFILLED,
    FETCH_PLAYERS_REJECTED
} from '../actions/PlayersActions';


// INITIALIZE STATE

const initialState = {
    players: [],
    fetching: false,
    fetched: false,
    failed: false
};


// REDUCER

export const FetchPlayersReducer = (state = initialState, action) => {
    switch(action.type) {
        case FETCH_PLAYERS_PENDING:
            return {
                ...state,
                players: [],
                fetching: true,
                fetched: false,
                failed: false
            };
        case FETCH_PLAYERS_FULFILLED:
            return {
                ...state,
                players: action.payload,
                fetching: false,
                fetched: true,
                failed: false
            };
        case FETCH_PLAYERS_REJECTED:
            return {
                ...state,
                players: [],
                fetching: false,
                fetched: false,
                failed: true
            };
        default:
            return state;
    }
};