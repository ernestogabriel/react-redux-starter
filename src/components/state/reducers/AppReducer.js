// IMPORT PACKAGE REFERENCES

import { combineReducers } from 'redux';


// IMPORT REDUCERS

import { FetchZipCodesReducer } from '../reducers/FetchZipCodesReducer';
import {FetchGroceryReducer, FetchGroceriesReducer } from '../reducers/FetchGroceriesReducer';


// EXPORT APP REDUCER

export const AppReducer = combineReducers({
    zipCodes: FetchZipCodesReducer,
    grocery: FetchGroceryReducer,
    groceries: FetchGroceriesReducer
});