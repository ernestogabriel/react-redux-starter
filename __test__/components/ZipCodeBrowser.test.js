import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import { ZipCodeBrowser } from '../../src/components/ZipCodeBrowser/ZipCodeBrowser';
import { initialState } from '../../src/components/state/reducers/FetchZipCodesReducer';

const mockStore = configureStore();
let store = null;

describe('<ZipCodeBrowser />', () => {
    describe('render()', () => {
        test('renders the component', () => {
            store = mockStore(initialState);
            const wrapper = shallow(<ZipCodeBrowser store={store} />);
            const component = wrapper.dive();

            expect(toJson(component)).toMatchSnapshot();
        })
    })
})