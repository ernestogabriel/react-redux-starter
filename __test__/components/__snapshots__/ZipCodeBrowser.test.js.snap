// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZipCodeBrowser /> render() renders the component 1`] = `
<div>
  <LoadingIndicator />
</div>
`;
